File: day05/part1.py
CODE_TRANS = str.maketrans({"F": "0", "B": "1", "L": "0", "R": "1"})


def compute(data):
    """
    >>> compute(["FBFBBFFRLR", "BFFFBBFRRR", "FFFBBBFRRR", "BBFFBBFRLL"])
    820
    """
    return max(int(code.translate(CODE_TRANS), 2) for code in data)


def main():
    import pathlib

    input_path = pathlib.Path(__file__).with_name("input.txt")

    with input_path.open() as f:
        print(compute(f.read().splitlines()))


if __name__ == "__main__":
    main()

File: day05/part1.exs
defmodule Day05.Part1 do
  def compute(data) do
    data
    |> String.trim()
    |> String.split("\n")
    |> Enum.map(fn code ->
      code
      |> String.replace(["F", "B", "L", "R"], fn
        "F" -> "0"
        "B" -> "1"
        "L" -> "0"
        "R" -> "1"
      end)
      |> String.to_integer(2)
    end)
    |> Enum.max()
  end

  def main do
    File.read!(Path.join(__DIR__, "input.txt"))
    |> compute
    |> IO.inspect()
  end
end

Day05.Part1.main()
